### PR TITLE
HttpResponseMessageExtensions fix & tests

### PR DESCRIPTION
**Pull request by bitai-cs**
Fix and tests for HttpResponseMessageExtensions.

## Summary by Sourcery

Adjust HTTP response message extension methods to use a custom client exception type, improve header/content parsing robustness, and add comprehensive test coverage.

Bug Fixes:
- Correct handling of success vs unsuccessful HTTP status codes in HttpReponseMessageExtensions.
- Fix media type detection and handling when response content or headers are null or empty.
- Ensure server header and date values are safely parsed and defaulted when missing or malformed.
- Properly distinguish middleware exception problem+json payloads from generic problem+json responses in a case-insensitive way.

Enhancements:
- Introduce a dedicated HttpClientExceptionResponse exception for HTTP client extension failures.
- Add helper methods for parsing Server headers and case-insensitive JSON property lookup in HTTP response processing.

Tests:
- Add extensive unit tests for HttpReponseMessageExtensions covering success, error, media type, header parsing, and edge-case behaviors.
- Relax selected string equality assertions in existing tests to use case-insensitive comparison.

### CODE REVIEW

File: tests/Bitai.WebApi.Helpers.Tests/HttpResponseMessageExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+
+        [Fact]
+        public async Task ToUnsuccessfulHttpResponseAsync_WithNoContent_ShouldReturnNoSuccessResponseWithEmptyContent()
+        {
+            // Arrange
+            var responseMessage = new HttpResponseMessage(HttpStatusCode.BadRequest)
+            {
+                Content = new StringContent(string.Empty)
+            };
+            responseMessage.Content.Headers.ContentType = null;
```
[the remark this answers is not in the document]
Comment (bitai-cs) [reply tests/Bitai.WebApi.Helpers.Tests/HttpResponseMessageExtensionsTests.cs:60]: I'll do it next version.

File: tests/Bitai.WebApi.Helpers.Tests/HttpResponseMessageExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+
+            // Act
+            var result = await responseMessage.ToUnsuccessfulHttpResponseAsync();
+
+            // Assert
+            var emptyResponse = Assert.IsType<NoSuccessResponseWithEmptyContent>(result);
+            Assert.Equal("(Server info not provided)", emptyResponse.WebServer);
+        }        
+
+        #endregion
```
[the remark this answers is not in the document]
Comment (bitai-cs) [reply tests/Bitai.WebApi.Helpers.Tests/HttpResponseMessageExtensionsTests.cs:196]: I'll review it the next version.

File: tests/Bitai.WebApi.Helpers.Tests/HttpResponseMessageExtensionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+            };
+
+            // Setup test serializer options
+            WebApiBaseClient.WebApiClientParameters.SerializerOptions = _testSerializerOptions;
+        }
+
+
+
+
+        #region ToUnsuccessfulHttpResponseAsync Tests
```
[the remark this answers is not in the document]
Comment (bitai-cs) [reply tests/Bitai.WebApi.Helpers.Tests/HttpResponseMessageExtensionsTests.cs:34]: I'll review it next version.
